core/validation/failures.py

```diff
@@ -19,120 +17,59 @@
         """Abstract method - implementations of this return message components.
 
-        :return: A tuple containing the sheet, subsection, and the message itself.
-        """
-
-
-class PreTransFormationFailure(ValidationFailure, ABC):
-    pass
-
-
-@dataclass
-class ExtraSheetFailure(ValidationFailure):
-    """Class representing an extra sheet failure."""
-
-    extra_sheet: str
-
-    def __str__(self):
-        """
-        Method to get the string representation of the extra sheet failure.
-        """
-        return (
-            "Extra Sheets Failure: The workbook included a sheet named"
-            f'"{self.extra_sheet}" but it is not in the schema.'
-        )
-
-    def to_message(self) -> tuple[str | None, str | None, str]:
-        raise UnimplementedErrorMessageException
-
-
-@dataclass
-class EmptySheetFailure(ValidationFailure):
-    """Class representing an empty sheet failure."""
-
-    empty_sheet: str
-
-    def __str__(self):
-        """Method to get the string representation of the empty sheet failure."""
-        return f'Empty Sheets Failure: The sheet named "{self.empty_sheet}" contains no ' "data."
-
-    def to_message(self) -> tuple[str | None, str | None, str]:
-        raise UnimplementedErrorMessageException
-
-
-@dataclass
-class ExtraColumnFailure(ValidationFailure):
-    """Class representing an extra column failure."""
-
-    sheet: str
-    extra_column: str
-
-    def __str__(self):
+        :return: A tuple containing the sheet, subsection, cell index and the message itself.
         """
-        Method to get the string representation of the extra column failure.
-        """
-        return f'Extra Column Failure: Sheet "{self.sheet}" Column' f' "{self.extra_column}" is not in the schema.'
-
-    def to_message(self) -> tuple[str | None, str | None, str]:
-        raise UnimplementedErrorMessageException
+        pass
 
 
 @dataclass
-class MissingColumnFailure(ValidationFailure):
-    """Class representing a missing column failure."""
+class NonHistoricalValidationFailure(ValidationFailure, ABC):
+    """Abstract base class representing a validation failure."""
 
     sheet: str
-    missing_column: str
-
-    def __str__(self):
-        """Method to get the string representation of the missing column failure."""
-        return (
-            f'Missing Column Failure: Sheet "{self.sheet}" Column'
-            f' "{self.missing_column}" is missing from the schema.'
-        )
-
-    def to_message(self) -> tuple[str | None, str | None, str]:
-        raise UnimplementedErrorMessageException
+    column: str | list[str]
+    row_index: int
 
 
 @dataclass
-class NonUniqueFailure(ValidationFailure):
-    """Class representing a non-unique value failure."""
+class GenericFailure(ValidationFailure):
+    """Class representing a sign-off failure in the Review & Sign-Off section."""
 
     sheet: str
-    column: str
+    section: str
+    cell_index: str
+    message: str
 
     def __str__(self):
```

Comment:
Every failure has these __str__() methods. But there functionality is eclipsed by to_message() is there anywhere on in the API where we would rather return __str__() or no?

Reply:
These could probably be improved but the reason is because we may want to use the `str` magic methods when developing new pipelines

Reply:
See slack for more info